ecl/eclagent/eclagent.cpp

```diff
@@ -1950,6 +1951,12 @@ void EclAgent::doProcess()
             break;
         }
 
+        if (w->getState() == WUStateCompleted)
+        {
+            const char * clustername = w->queryClusterName();
+            if (getClusterTypeByClusterName(clustername)==ThorLCRCluster)
+                analyseWorkunit(static_cast<IConstWorkUnit*>(w.get()));
```

Comment:
Minor: would be better to pass in w as an IWorkUnit * - then there would be no need for the code to lock the workunit.  (There also shouldn't need to be a cast to convert a IWorkUnit * to a IConstWorkUnit *.